src/rules/__tests__/prefer-mocked.test.ts

```diff
@@ -0,0 +1,407 @@
+import dedent from 'dedent';
+import rule from '../prefer-mocked';
+import { FlatCompatRuleTester as RuleTester } from './test-utils';
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('@typescript-eslint/parser'),
+});
+
+ruleTester.run('prefer-mocked', rule, {
```

Comment:
nit: generally I'd prefer to not use `dedent` for single line calls, and not have a newline between cases